pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -418,6 +420,33 @@ func buildInboundLocalityLbEndpoints(bind string, port int) []endpoint.LocalityL
 	}
 }
 
+func generateInboundPassthroughClusters(env *model.Environment) []*apiv2.Cluster {
+	clusters := make([]*apiv2.Cluster, 0, 2)
+	inboundPassthroughClusterIpv4 := buildDefaultPassthroughCluster(env)
+	inboundPassthroughClusterIpv4.Name = util.InboundPassthroughClusterIpv4
+	inboundPassthroughClusterIpv4.UpstreamBindConfig = &core.BindConfig{
+		SourceAddress: core.SocketAddress{
+			Address: util.InboundPassthroughBindIpv4,
+			PortSpecifier: &core.SocketAddress_PortValue{
+				PortValue: uint32(0),
+			},
+		},
+	}
+
+	inboundPassthroughClusterIpv6 := buildDefaultPassthroughCluster(env)
+	inboundPassthroughClusterIpv6.Name = util.InboundPassthroughClusterIpv6
+	inboundPassthroughClusterIpv6.UpstreamBindConfig = &core.BindConfig{
+		SourceAddress: core.SocketAddress{
+			Address: util.InboundPassthroughBindIpv6,
+			PortSpecifier: &core.SocketAddress_PortValue{
+				PortValue: uint32(0),
+			},
+		},
+	}
+	clusters = append(clusters, inboundPassthroughClusterIpv4, inboundPassthroughClusterIpv6)
+	return clusters
```

Comment:
nit: you could just do `return []*apiv2.Cluster{inboundPassthroughClusterIpv4, inboundPassthroughClusterIpv6}`

Reply:
Great catch! Besides it I erase the clusters var. See update